### ISSUE TITLE
Add authentication

### ISSUE THREAD

**Issue by watzon** [rjframe/json-rpc#8]
Several RPC servers require authentication to access them, such as is the case with litecoind and similar. It would be nice if you could do something like `client.basicAuth("username", "password")`, and have it generate the necessary authentication header.

**Comment by rjframe** [comment]
The server needs to be able to support requiring clients to authenticate too; currently an HTTP transport still needs to be written before either can be done (basic auth would be an HTTP transport-level authentication).

Thanks.

**Comment by watzon** [comment]
Yeah I figured as much, but I figured I'd get the issue listed here. Great project.
